compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java

```diff
@@ -53,10 +56,34 @@ static CompilerPluginManager from(PackageCompilation compilation) {
         DependencyGraph<ResolvedPackageDependency> dependencyGraph = packageResolution.dependencyGraph();
         List<Package> directDependencies = getDirectDependencies(rootPkgNode, dependencyGraph);
         List<CompilerPluginInfo> compilerPlugins = loadEngagedCompilerPlugins(directDependencies);
+        List<CompilerPluginInfo> inBuiltCompilerPlugins = loadInBuiltCompilerPlugins(rootPkgNode.packageInstance());
+        compilerPlugins.addAll(inBuiltCompilerPlugins);
         List<CompilerPluginContextIml> compilerPluginContexts = initializePlugins(compilerPlugins);
         return new CompilerPluginManager(compilation, compilerPluginContexts);
     }
 
+    private static List<CompilerPluginInfo> loadInBuiltCompilerPlugins(Package rootPackage) {
```

Comment:
I can suggest the following changes. 
1) The built-in compiler plugins do not change across compilations, but package-provided plugins may change per compilation. Therefore we could load and cache built-in compiler plugins. Use plugins from the cache in subsequent compilations. 
2) Move the class loading logic to the `CompilerPlugins.java` class. You can cache the built-in plugin list in that class. Perhaps you could load the plugins in a static block even.

Reply:
Addressed in 40a3bf6